opencraft/utils.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+General utility functions for OCIM.
+"""
+from typing import Any, Dict, Iterable
+
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.template.loader import get_template
+
+
+def get_site_url(relative_path: str = '') -> str:
+    """
+    Uses the site framework to build an absolute URL.
+
+    This function can be used to build an absolute URL for the site without
+    needing access to the request object.
+    """
+    domain = get_current_site(request=None).domain
+    if settings.DEBUG:
+        scheme = "http"
+    else:
+        scheme = "https"
+    return f"{scheme}://{domain}{relative_path}"
+
+
+def build_email_context(context=None):
+    """
+    Builds a context object with common settings for emails.
+    """
```

Comment:
Please add a docstring about `context` and information that it overrides generated values.